src/util/performance.js

```diff
@@ -1,45 +1,74 @@
 // @flow
 
+import window from '../util/window';
 import type {RequestParameters} from '../util/ajax';
 
-// Wraps performance to facilitate testing
-// Not incorporated into browser.js because the latter is poisonous when used outside the main thread
-const performanceExists = typeof performance !== 'undefined';
-const wrapper = {};
+const performance = window.performance;
 
-wrapper.getEntriesByName = (url: string) => {
-    if (performanceExists && performance && performance.getEntriesByName)
-        return performance.getEntriesByName(url);
-    else
-        return false;
-};
-
-wrapper.mark = (name: string) => {
-    if (performanceExists && performance && performance.mark)
-        return performance.mark(name);
-    else
-        return false;
-};
-
-wrapper.measure = (name: string, startMark: string, endMark: string) => {
-    if (performanceExists && performance && performance.measure)
-        return performance.measure(name, startMark, endMark);
-    else
-        return false;
-};
+export type PerformanceMetrics = {
+    loadTime: number,
+    fullLoadTime: number,
+    fps: number,
+    percentDroppedFrames: number
+}
 
-wrapper.clearMarks = (name: string) => {
-    if (performanceExists && performance && performance.clearMarks)
-        return performance.clearMarks(name);
-    else
-        return false;
+export const PerformanceMarkers = {
+    create: 'create',
+    load: 'load',
+    fullLoad: 'fullLoad'
 };
 
-wrapper.clearMeasures = (name: string) => {
-    if (performanceExists && performance && performance.clearMeasures)
-        return performance.clearMeasures(name);
-    else
-        return false;
+let lastFrameTime = null;
+let frameTimes = [];
+
+const minFramerateTarget = 30;
```

Comment:
What are the tradeoffs between using 30 and 60 here?

Reply:
Setting it at seemed to introduce a lot of noise due to a combination of  measurement error and being very rarely always locked at 60, so almost every frame ends up missing the target. 
The idea is to capture `really bad` frames in a metric, and half of our ideal seemed like a good start.